src/Liquid.Messaging/Interfaces/ILiquidProducer.cs

```diff
@@ -22,5 +23,14 @@ public interface ILiquidProducer<TEntity>
         /// <param name="messageBody">Body of message to be sent.</param>
         /// <param name="customProperties">Message header properties.</param>
         Task SendMessageAsync(TEntity messageBody, IDictionary<string, object> customProperties = null);
+
+        /// <summary>
+        /// Structures and processes sending a <typeparamref name="TEntity"/> message with
+        /// its headers <paramref name="customProperties"/>.
+        /// </summary>
+        /// <param name="scheduleEnqueueTimeUtc">Scheduled Time to Enqueue a message.</param>
+        /// <param name="messageBody">Body of message to be sent.</param>
+        /// <param name="customProperties">Message header properties.</param>
+        Task ScheduleMessageAsync(DateTimeOffset scheduleEnqueueTimeUtc, TEntity messageBody, IDictionary<string, object> customProperties = null);
```

Comment:
this class is the base interface for all adapters (cartridges) implementations, and it's methods must be common features of principal messaging vendors (kafka, rabbit, event hub, etc). Please check if this is a common feature, and if it's not, I sugest you implement a new interface for , using some kind of interface composure.